code/game/machinery/buttons.dm

```diff
@@ -140,12 +140,15 @@
 	if(device)
 		to_chat(user, span_warning("The button already contains a device!"))
 		return ITEM_INTERACT_BLOCKING
+	if(!(new_device.assembly_behavior & ASSEMBLY_FUNCTIONAL_OUTPUT))
+		to_chat(user, span_warning("\The [new_device] won't really do anything meaningful inside of the button..."))
+		return ITEM_INTERACT_BLOCKING
 	if(!user.transferItemToLoc(new_device, src, silent = FALSE))
 		to_chat(user, span_warning("\The [new_device] is stuck to you!"))
 		return ITEM_INTERACT_BLOCKING
 
 	device = new_device
-	SEND_SIGNAL(new_device, COMSIG_ASSEMBLY_ADDED_TO_BUTTON, src, user)
+	SEND_SIGNAL(new_device, COMSIG_ASSEMBLY_ADDED_TO_OBJECT, src, user)
```

Comment:
The common pattern for signals is that they should be sent from 1 point in code. If you need to send a signal from multiple points in code, you need to use different signals.
For example, this would stay as COMSIG_ASSEMBLY_ADDED_TO_BUTTON